walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,46 +26,106 @@
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
     Set<String> attendees =  new HashSet<String>();
+    Set<String> optionalAttendees =  new HashSet<String>();
     int meetingDur = (int) request.getDuration();
     attendees.addAll(request.getAttendees());
-    if (meetingDur > TimeRange.WHOLE_DAY.duration()){
+    optionalAttendees.addAll(request.getOptionalAttendees());
+    if (meetingDur > TimeRange.WHOLE_DAY.duration()) {
       return possibleTimes;
     }
-    if (events.isEmpty() || attendees.isEmpty()){
+    if (events.isEmpty()) {
       possibleTimes.add(TimeRange.WHOLE_DAY);
       return possibleTimes;
     }
-    ArrayList<TimeRange> busyWithOverlap = getBusyTimes(events, attendees);
-    possibleTimes = findFreeTimes(busyWithOverlap, request.getDuration());
-    return possibleTimes;
+    return getPossibleTimes(events, attendees, optionalAttendees, request.getDuration());
+    
   }
 
-/**
-  * Returns an ArrayList of meeting times that are unavailable due to schedule
-  * conflicts and overlapping meeting times within the given group of attendees. 
+  /**
+  * Returns an ArrayList of all possible meeting times that are free for the given list of attendees,
+  * and includes optional attendees if possible. 
+  *
+  * @param  events the Collection of Event objects
+  * @param  requestAttendees the Set of mandatory attendees
+  * @param  optAttendees the Set of optional attendees
+  * @param  duration the length of the requested meeting
+  * @return    the ArrayList of all possible meeting times that work with the attendees schedules
+  */
+  private ArrayList<TimeRange> getPossibleTimes(Collection<Event> events, Set<String> mandatoryAttendees,
+          Set<String> optAttendees, long duration) {
+    ArrayList<TimeRange> mandatoryBusyTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalBusyTimes = new ArrayList<TimeRange>();
+    mandatoryBusyTimes = getBusyTimes(events, mandatoryAttendees);
+    optionalBusyTimes = getBusyTimes(events, optAttendees);
```

Comment:
No need to declare these just to reassign them
`ArrayList<TimeRange> mandatoryBusyTimes =  getBusyTimes(eents, mandatoryAttendees);` should work fine

Reply:
Got it!